src/ThemeMeUp.Mobile/ThemeMeUp.Mobile/ViewModels/MainPageViewModel.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+using ThemeMeUp.Core.Boundaries.GetLatestWallpapers;
+using ThemeMeUp.Core.Entities;
+using Xamarin.CommunityToolkit.ObjectModel;
+
+namespace ThemeMeUp.Mobile.ViewModels
+{
+    public class MainPageViewModel : BaseViewModel
+    {
+
+        private readonly IGetLatestWallpapersUseCase _useCase;
+        private readonly LatestWallpaperPresenter _presenter;
+
+        #region Commands
+
+        public IAsyncCommand RefreshCommand { get; }
+
+        #endregion
+
+        public MainPageViewModel(IGetLatestWallpapersUseCase useCase, LatestWallpaperPresenter presenter)
+        {
+            _useCase = useCase;
+            _presenter = presenter;
+
+            Title = "Theme Me Up";
+
+            RefreshCommand = new AsyncCommand(RefreshAsync, CanExecute);
+
+            Wallpapers = new ObservableCollection<Wallpaper>();
+        }
+
+        private async Task RefreshAsync()
+        {
+            try
+            {
+                IsRefreshing = true;
+
+                Wallpapers.Clear();
+                await GetLatestWallpapersAsync();
+            }
+            finally
+            {
+                IsRefreshing = false;
+            }
+           
+        }
+
+        public async Task GetLatestWallpapersAsync()
+        {
+            try
+            {
+                IsBusy = true;
+
+                await Task.Run(async () =>
+                {
+                    await _useCase.Execute(new GetLatestWallpapersInput());
+                });
```

Comment:
Is there any specific reason why we wrap the task? 🤔 

Couldn't it be just the following?

```cs
await _useCase.Execute(new GetLatestWallpapersInput());
```

Reply:
Does the method run on the main thread if we use:
```cs
await _useCase.Execute(new GetLatestWallpapersInput());
```

Android shows a "Close app" prompt if the main thread is busy for longer then 5 seconds (I added that so the api call doesnt happen on the Main Thread).

Reply:
> does the method run on the main thread if we use:

It shouldn't no.
The whole point of await is to prevent your code locking up the UI thread.
As soon as you await it, the code runs asyncronously and won't block.

Reply:
Awesome thanks, I'm going to remove the wrapper stuff.